### PR TITLE
Improve Plugin documentation

### PR DESCRIPTION
**Pull request by janpio**
Plugin Development Guide:
- fix git repo url for `cordova plugin add` description
- add npm package name for `cordova plugin add` description
- change explanation of plugin.xml content to list
- fix description of plugin.id
- add several internal links
- move "Validating a Plugin using Plugman" to a better location
- fix formatting

Android Plugin Development Guide:
- Restructure introduction (slightly)
- `src/android/Echo.java` file indentation
- format parameter in text as code

iOS Plugin Development Guide:
- Restructure introduction (slightly)

### CODE REVIEW

**Comment by timbru31** [comment]
Thanks for updating the documentation! Getting started with plugin development was really a pain so far ;)

**Comment by janpio** [comment]
I feel you, just going through the process myself again and taking the opportunity to make this a bit less terrible.

But sounds like you did the same @timbru31 - do you have any stuff to add here?

TODO from Slack:
- `--link` (added via #975) only applies to native parts of the plugin (not package.json, .js etc) and so is only sometimes useful ("the only thing is that changes in javascript, types, plugin.xml or package.json are not propagated, just the native files")
- `--link` workflow: create the project, add the plugin via `--link`, then run with xcode/androidstudio, modify native parts of the plugin in the plugin source folder, re-run.

**Comment by timbru31** [comment]
> do you have any stuff to add here?

Sure. Will add some input later :)

File: www/docs/en/dev/guide/platforms/windows/plugin.md [hunk 1]
```diff
@@ -66,11 +68,26 @@ cordova.commandProxy.add("Echo",{
         }
     }
 });
+
+// or alternative syntax
```
Comment (janpio) [inline www/docs/en/dev/guide/platforms/windows/plugin.md]: No idea if this actually works - will have to try it and compare that to the currently documented implementation.

Comment (janpio) [reply www/docs/en/dev/guide/platforms/windows/plugin.md]: Those are probably both actually calling https://github.com/apache/cordova-js/blob/a770d95f129f0e16c696bcecd00dc8d3e8e53b5c/src/common/exec/proxy.js

**Comment by janpio** [comment]
TODO

"Plugin Exec Proxy" form the WIndows plugin docs also applies to browser and can be used for all - maybe move description to general plugin docs file

**Comment by erisu** [comment]
@timbru31 It looks like you were going to leave some input? If not and if you feel this is OK as it stands, I like to merge it in.

**Comment by timbru31** [comment]
I’ll add this to my to-do list for tomorrow. If I fail to do so, feel free to merge on Monday

**Pull request merged**